.github/workflows/check.yml

```diff
@@ -5,101 +5,106 @@ env:
 
 jobs:
   check:
-    name: Neqo Build and Test
-    runs-on: ubuntu-22.04
-    env:
-      RUSTFLAGS: -C link-arg=-fuse-ld=lld
+    name: Build & test
     strategy:
+      fail-fast: false
       matrix:
-        rust-toolchain: [stable, 1.65.0]
+        os: [ubuntu-latest, macos-latest] #, windows-latest]
+        rust-toolchain: [1.65.0, stable] #, beta]
```

Comment:
@mb wanted beta as an early warning for new lints.  We get less value from having the full matrix there, but given that we've seen errors in Mac builds, it might be OK to just do the full set.

Reply:
I put `beta` there as a reminder how to figure out how to only make it run informationally (i.e., have fails not be critical).